java/src/Ice/src/main/java/com/zeroc/Ice/ValueFactoryManagerI.java

```diff
@@ -2,15 +2,12 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-package com.zeroc.IceInternal;
+package com.zeroc.Ice;
 
-import com.zeroc.Ice.ValueFactory;
-
-final class ValueFactoryManagerI implements com.zeroc.Ice.ValueFactoryManager {
+final class ValueFactoryManagerI implements ValueFactoryManager {
     public synchronized void add(ValueFactory factory, String id) {
-        Object o = _factoryMap.get(id);
-        if (o != null) {
-            throw new com.zeroc.Ice.AlreadyRegisteredException("value factory", id);
```

Comment:
One of the few places where I actually changed the code. I had to fix all the uses of `Object` since they changed from meaning `java.lang.Object` to `com.zeroc.Ice.Object`.

But this is literally just a poor-man's `containsKey`... so I used that instead.